iohub/ngff.py

```diff
@@ -900,6 +907,47 @@ def update_channel(self, chan_name: str, target: str, data: ArrayLike):
         ortho_sel[ch_ax] = ch_idx
         img.set_orthogonal_selection(tuple(ortho_sel), data)
 
+    def initialize_pyramid(self, levels: int) -> None:
+        """
+        Initializes the pyramid arrays with a down scaling of 2 per level.
+        Decimals shapes are rounded up to ceiling.
+        Scales metadata are also updated.
+
+        Parameters
+        ----------
+        levels : int
+            Number of down scaling levels, if levels is 1 nothing happens.
+        """
+        array = self.data
+        for level in range(1, levels):
+            factor = 2**level
```

Comment:
Do we need a check that factor is smaller than shape?

Reply:
Hi @ziw-liu , I didn't see this comment when merging the PR. We get the ceiling of the `shape / factor`, so it should be fine. Thanks for pointing this out.